tools/nixpkgs-hammer

```diff
@@ -141,37 +229,52 @@ def main(args):
-                      {{
-                        report = [ {{
-                          name = "EvalError";
-                          msg = "Cannot evaluate attribute ‘{attr}’ in ‘${{packageSet}}’.";
-                          severity = "warning";
-                          link = false;
-                        }} ];
-                        position = null;
-                      }}
-                    else if result.value == null then
-                      {{
-                        report = [ {{
-                          name = "AttrPathNotFound";
-                          msg = "Packages in ‘${{packageSet}}’ do not contain ‘{attr}’ attribute.";
-                          severity = "error";
-                          link = false;
-                        }} ];
-                        position = null;
-                      }}
-                    else
-                      {{
-                        report = if maybeReport.success then maybeReport.value else [];
-                        position = result.value.meta.position;
-                      }};
-                """
+            "{attr}" =
+              let
+                result = builtins.tryEval pkgs.{escape_attr(attr)} or null;
+                maybeReport = builtins.tryEval (result.value.__nixpkgs-hammering-state.reports or []);
+              in
+                if !(result.success && maybeReport.success) then
+                  {{
+                    report = [ {{
+                      name = "EvalError";
+                      msg = "Cannot evaluate attribute ‘{attr}’ in ‘${{packageSet}}’.";
+                      severity = "warning";
+                      link = false;
+                    }} ];
+                    location = null;
+                    outputPath = null;
+                    drvPath = null;
+                  }}
+                else if result.value == null then
+                  {{
+                    report = [ {{
+                      name = "AttrPathNotFound";
+                      msg = "Packages in ‘${{packageSet}}’ do not contain ‘{attr}’ attribute.";
+                      severity = "error";
+                      link = false;
+                    }} ];
+                    location = null;
+                    outputPath = null;
+                    drvPath = null;
+                  }}
+                else
+                  {{
+                    report = if maybeReport.success then maybeReport.value else [];
+                    location = let
+                      position = result.value.meta.position or null;
+                      posSplit = if (builtins.isNull position) then
+                        [null null null]
+                      else
+                        builtins.split ":" result.value.meta.position;
+                    in {{
+                      file = builtins.elemAt posSplit 0;
+                      line = builtins.elemAt posSplit 2;
+                    }};
```

Comment:
We could just do

```suggestion
                      posSplit = builtins.split ":" result.value.meta.position;
                    in
                      if position == null then
                        null
                      else {{
                        file = builtins.elemAt posSplit 0;
                        line = builtins.elemAt posSplit 2;
                      }};
```

so that we always have `location == null || (builtins.isString location.file && builtins.isInt location.line)`.